src/SFA.DAS.AANHub.Api.UnitTests/Controllers/MembersControllerTests/MembersControllerGetMemberActivitiesTests.cs

```diff
@@ -0,0 +1,166 @@
+        GetMemberActivitiesResult getMemberActivitiesResult)
+    {
+        // Arrange
+        var response = new ValidatedResponse<GetMemberActivitiesResult>(getMemberActivitiesResult);
+        mediatorMock.Setup(m => m.Send(It.Is<GetMemberActivitiesQuery>(q => q.MemberId == memberId), It.IsAny<CancellationToken>()))
+            .ReturnsAsync(response);
+
+        // Act
+        var result = await sut.GetMemberActivities(memberId);
+
+        // Assert
+        result.As<OkObjectResult>().Should().NotBeNull();
+        result.As<OkObjectResult>().Value.Should().Be(getMemberActivitiesResult);
+    }
+
+    [Test, AutoData]
+    public async Task GetMemberActivities_HandlerReturnsData_ShouldReturnExpectedValue(
+    GetMemberActivitiesResult getMemberActivitiesResult)
+    {
+        // Arrange
+        var response = new ValidatedResponse<GetMemberActivitiesResult>(getMemberActivitiesResult);
+        mediatorMock.Setup(m => m.Send(It.Is<GetMemberActivitiesQuery>(q => q.MemberId == memberId), It.IsAny<CancellationToken>()))
+            .ReturnsAsync(response);
+
+        // Act
+        var result = await sut.GetMemberActivities(memberId);
+        var objectResult = result as OkObjectResult;
+        var memberActivitiesResult = (GetMemberActivitiesResult)objectResult!.Value!;
+
+        // Assert
+        Assert.Multiple(() =>
+        {
+            Assert.That(memberActivitiesResult, Is.Not.Null);
+            Assert.That(memberActivitiesResult.LastSignedUpDate, Is.EqualTo(getMemberActivitiesResult.LastSignedUpDate));
+            Assert.That(memberActivitiesResult.EventsPlanned, Is.Not.Null);
+            Assert.That(memberActivitiesResult.EventsPlanned.EventsDateRange, Is.EqualTo(getMemberActivitiesResult.EventsPlanned.EventsDateRange));
+            Assert.That(memberActivitiesResult.EventsPlanned.EventsDateRange.FromDate, Is.EqualTo(getMemberActivitiesResult.EventsPlanned.EventsDateRange.FromDate));
+            Assert.That(memberActivitiesResult.EventsPlanned.EventsDateRange.ToDate, Is.EqualTo(getMemberActivitiesResult.EventsPlanned.EventsDateRange.ToDate));
+            Assert.That(memberActivitiesResult.EventsPlanned.Events.Count, Is.EqualTo(getMemberActivitiesResult.EventsPlanned.Events.Count));
+            Assert.That(memberActivitiesResult.EventsAttended, Is.Not.Null);
+            Assert.That(memberActivitiesResult.EventsAttended.EventsDateRange, Is.EqualTo(getMemberActivitiesResult.EventsAttended.EventsDateRange));
+            Assert.That(memberActivitiesResult.EventsAttended.EventsDateRange.FromDate, Is.EqualTo(getMemberActivitiesResult.EventsAttended.EventsDateRange.FromDate));
+            Assert.That(memberActivitiesResult.EventsAttended.EventsDateRange.ToDate, Is.EqualTo(getMemberActivitiesResult.EventsAttended.EventsDateRange.ToDate));
+            Assert.That(memberActivitiesResult.EventsAttended.Events.Count, Is.EqualTo(getMemberActivitiesResult.EventsAttended.Events.Count));
+        });
```

Comment:
This is pointless isn't it? All the tests post from top two and last one seems pointless as far as I can see.

Reply:
Done